xwayland/xwm.c

```diff
@@ -940,6 +941,8 @@ static void xwm_handle_configure_notify(struct wlr_xwm *xwm,
 	xsurface->width = ev->width;
 	xsurface->height = ev->height;
 
+    wlr_signal_emit_safe(&xsurface->events.configure_notify, xsurface);
```

Comment:
I'd prefer to trigger this event after the event has been completely handled by wlroots, ie. at the end of the function after the `override_redirect` handling. Otherwise the compositor will see half-updated state.